src/gurt/examples/telem_producer_example.c

```diff
@@ -0,0 +1,486 @@
+ * in some location other than exactly where it is being used.
+ *
+ * \param[in]	node_list	A list of metrics that were previously added
+ */
+void use_manually_added_metrics(struct d_tm_nodeList_t *node_list)
+{
+	int	rc;
+
+	while (node_list) {
+		switch (node_list->dtnl_node->dtn_type) {
+		case D_TM_DIRECTORY:
+			break;
+		case D_TM_COUNTER:
+			/**
+			 * Supplying an initialized pointer to the metric
+			 * so it is only used and not created implicitly here.
+			 */
+			rc = d_tm_increment_counter(&node_list->dtnl_node,
+						    NULL);
+			if (rc != D_TM_SUCCESS) {
+				printf("d_tm_increment_counter failed, "
+				       "rc = %d\n", rc);
+			}
+			break;
+		default:
+			printf("Item: %s has unknown type: 0x%x\n",
+			       node_list->dtnl_node->dtn_name,
+			       node_list->dtnl_node->dtn_type);
+			break;
+		}
+		node_list = node_list->dtnl_next;
+	}
+}
+
+int
+main(int argc, char **argv)
+{
+	static struct d_tm_node_t	*entry;
+	static struct d_tm_node_t	*loop;
+	static struct d_tm_node_t	*timer1;
+	static struct d_tm_node_t	*timer2;
+	struct d_tm_nodeList_t		*node_list;
+	struct d_tm_nodeList_t		*head;
+	int 				rc;
+	int 				simulated_rank = 0;
+	int 				i;
+
+	if (argc < 2) {
+		printf("Specify an integer that identifies this producer's "
+		       "rank.  Specify the same value to the consumer.\n");
+		exit(0);
+	}
+
+	simulated_rank = atoi(argv[1]);
+	printf("This simulated rank has ID: %d\n", simulated_rank);
+
+	/**
+	 * Call d_tm_init() only once per process,
+	 * i.e. in iosrv/init.c/server_init()
+	 */
+	rc = d_tm_init(simulated_rank, D_TM_SHARED_MEMORY_SIZE);
+	if (rc != 0) {
+		goto failure;
+	}
+
+	/**
+	 * The API is ready to use.  Add a counter that will be identified in
+	 * the tree by this file name, function name, and the name "sample
+	 * counter", i.e.:
+	 * "src/gurt/examples/telem_producer_example.c/main/sample counter"
+	 *
+	 * On the first call through, the pointer to this metric is NULL, and
+	 * the API looks up the metric by name.  It won't find it, so it creates
+	 * it.  The counter is created, and incremented by one.  It now has the
+	 * value 1.
+	 */
+	rc = d_tm_increment_counter(&entry, __FILE__, __func__,
+				    "sample counter", NULL);
+	if (rc != D_TM_SUCCESS) {
+		printf("d_tm_increment_counter failed, rc = %d\n", rc);
+		goto failure;
+	}
+
+	/**
+	 * Increment another counter in a loop.
+	 * On the first iteration, the API finds the metric by name and
+	 * initializes 'loop'.  On subsequent iterations, the pointer is used
+	 * for faster lookup.
+	 */
+	for (i = 0; i < 1000; i++) {
+		rc = d_tm_increment_counter(&loop, __FILE__, __func__,
+					    "loop counter", NULL);
+		if (rc != D_TM_SUCCESS) {
+			printf("d_tm_increment_counter failed, rc = %d\n", rc);
+			goto failure;
+		}
+	}
+
+	/**
+	 * How long does it take to execute test_function()?
+	 * When the duration timer is created, specify the clock type from:
+ 	 * D_TM_CLOCK_REALTIME which is CLOCK_REALTIME
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs